### ISSUE TITLE
Issues attaching to USDT probes in containers

### ISSUE THREAD

**Issue by dalehamel** [iovisor/bpftrace#1071]
Similar to https://github.com/iovisor/bpftrace/issues/968, as expected USDT probes in containers don't work.

I'm mostly writing this issue as a reminder to myself to fix this, similarly to how it was done in #1021 

I debugged this a little bit with a colleague today (@clandry94) trying to attach to USDT probes on a JVM process inside of a container.

After modifying the semantic analyzer to pass a pid to `resolve_binary_path`, we still got the error:

```
 ERROR: usdt target file '/proc/3509211/root/usr/share/elasticsearch/jdk/lib/server/libjvm.so' does not exist or is not executable
```

Despite the file very much existing and being executable. It even shows up when you list the USDT probes for the process.

I don't expect this to be very complicated to fix, and we confirmed that it was working in bpftrace 0.9.2 (but not master), so this is clearly a regression. A regression runtime test will need to be added to prevent this from regressing again once fixed.

**Comment by dalehamel** [comment]
with https://github.com/iovisor/bcc/pull/2710 merged, part of this underlying problem is fixed.

To caveat this, I discovered that:

* The bug is only on a codepath in BCC that older kernels will use, in my case 4.14 was using this code path. Newer kernels (5.0.0, possibly earlier) use a different code path and aren't susceptible to the bug in attaching to probes
* There is a separate bug that i discovered where resolving probes on overlayfs or btrfs requires special handling by setting a `kludge` bit, saying to match by inode only and not also device maj/min. I filed this as https://github.com/iovisor/bcc/issues/2715 to discuss a solution to
